Dockerfile

```diff
@@ -1,10 +1,61 @@
-FROM golang:1.15-alpine AS build
-RUN apk add --no-cache git make jq-dev gcc libc-dev oniguruma-dev bash
+# syntax=docker/dockerfile:1
 
-WORKDIR /go/src/github.com/jzelinskie/faq
-COPY . .
+ARG GO_VERSION=1.17
+ARG GORELEASER_XX_VERSION=1.2.5
 
-RUN make install FAQ_LINK_STATIC=true
+FROM --platform=$BUILDPLATFORM crazymax/goreleaser-xx:${GORELEASER_XX_VERSION} AS goreleaser-xx
+FROM --platform=$BUILDPLATFORM crazymax/goxx:${GO_VERSION} AS base
+COPY --from=goreleaser-xx / /
+RUN apt-get update && apt-get install --no-install-recommends -y git
+WORKDIR /src
+
+FROM base AS vendored
+RUN --mount=type=bind,target=.,rw \
+  --mount=type=cache,target=/go/pkg/mod \
+  go mod tidy && go mod download
+
+FROM base AS lint
+RUN apt-get install --no-install-recommends -y gcc libc6-dev libjq-dev libonig-dev
+RUN go install golang.org/x/lint/golint@latest
+RUN --mount=type=bind,target=. \
+  --mount=type=cache,target=/root/.cache \
+  golint ./...
+
+FROM vendored AS test
+RUN apt-get install --no-install-recommends -y gcc libc6-dev libjq-dev libonig-dev
+RUN --mount=type=bind,target=. \
+  --mount=type=cache,target=/go/pkg/mod \
+  --mount=type=cache,target=/root/.cache \
+  go test -v -coverprofile=/tmp/coverage.txt -covermode=atomic -race ./...
+
+FROM scratch AS test-coverage
+COPY --from=test /tmp/coverage.txt /coverage.txt
+
+FROM vendored AS build
+ARG TARGETPLATFORM
+ENV CGO_ENABLED=1
+ENV OSXCROSS_MP_INC=1
+RUN --mount=type=cache,sharing=private,target=/var/cache/apt \
+  --mount=type=cache,sharing=private,target=/var/lib/apt/lists \
+  goxx-apt-get install -y binutils gcc g++ pkg-config libjq-dev libonig-dev
+RUN goxx-macports install jq
```

Comment:
Here is the magic section in the Dockerfile to handle the specific toolchain packages based on the `TARGETPLATFORM` value. More info about the `goxx-*` wrappers: https://github.com/crazy-max/goxx#wrappers

Reply:
For windows I will work on a cygwin port wrapper. https://github.com/crazy-max/goxx/issues/14